src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,124 @@
+            if (!nextLineStation.isPresent()) {
+                break;
+            }
+            downStation = nextLineStation.get().getDownStation();
+            stations.add(downStation);
+        }
+        return stations;
+    }
+
+    private Optional<Section> findUpToDownConnected(Station finalDownStation) {
+        return sections.stream()
+                .filter(it -> it.getUpStation() == finalDownStation)
+                .findFirst();
+    }
+
+    private Station findUpStation() {
+        Station downStation = sections.stream()
+                .findFirst()
+                .map(Section::getUpStation)
+                .orElse(null);
+
+        while (downStation != null) {
+            Station finalDownStation = downStation;
+            Optional<Section> nextLineStation = sections.stream()
+                    .filter(it -> it.getDownStation() == finalDownStation)
+                    .findFirst();
+            if (!nextLineStation.isPresent()) {
+                break;
+            }
+            downStation = nextLineStation.get().getUpStation();
+        }
+        return downStation;
+    }
+
+    public void removeSection(Line line, Station targetStation) {
+        if (sections.size() <= MINIMUM_REMOVAL_LIMIT) {
+            throw new IllegalStateException(REMOVAL_NOT_ALLOWED);
+        }
+        Optional<Section> upwardSection = findSectionWithUpStation(targetStation);
+        Optional<Section> downwardSection = findSectionWithDownStation(targetStation);
+
+        if (upwardSection.isPresent() && downwardSection.isPresent()) {
+            Station newUpStation = downwardSection.get().getUpStation();
+            Station newDownStation = upwardSection.get().getDownStation();
+            int newDistance = upwardSection.get().getDistance() + downwardSection.get().getDistance();
+            sections.add(Section.of(line, newUpStation, newDownStation, newDistance));
+        }
+
+        upwardSection.ifPresent(sections::remove);
+        downwardSection.ifPresent(sections::remove);
+    }
+
+    public void addSection(Line line, Station upStation, Station downStation, int distance) {
+        List<Station> stations = getStations();
+        validateStations(stations, upStation, downStation);
+
+        if (stations.contains(upStation)) {
+            findSectionWithUpStation(upStation)
+                    .ifPresent(s -> s.updateUpStation(downStation, distance));
+        }
+        if (stations.contains(downStation)) {
+            findSectionWithDownStation(downStation)
+                    .ifPresent(s -> s.updateDownStation(upStation, distance));
+        }
+        sections.add(Section.of(line, upStation, downStation, distance));
+    }
+
+    private void validateStations(List<Station> stations, Station upStation, Station downStation) {
+        if (stations.size() > 0 && stations.contains(upStation) && stations.contains(downStation)) {
```

Comment:
아주 소소한 의견이지만 `stations.size() > 0 && stations.contains(upStation) && stations.contains(downStation)` 부분들도 `method`로 추출 되면 코드 가독성 측면에서 좀 더 개선해 볼 수 있을 것 같습니다 😄 
관점에 따라서는 과도한 추출로도 보여질 수 있으니 
이 부분도 개인적 의견입니다 😄 
참고만 해주시면 감사하겠습니다 🙇 

```suggestion
        if hasDuplicatedStation(stations, upStation, downStation) {
```

Reply:
같은 고민을 하게되서 기쁩니다. 
처음에 메서드 추출했다가 upstation, downstation 갖고 있는 것을 바로 보여주는게 더 낫지 않나 싶어, 원상태로 두었습니다.